tracer/src/Datadog.Trace/Util/UnsafeHelper.cs

```diff
@@ -0,0 +1,56 @@
+﻿// <copyright file="UnsafeHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System.Reflection.Emit;
+using System.Runtime.CompilerServices;
+
+namespace Datadog.Trace.Util;
+
+internal static class UnsafeHelper
```

Comment:
We recently vendored in a bunch of `Unsafe` code from the .NET runtime - I wonder if we could/should use that instead?

Reply:
good point! changed in https://github.com/DataDog/dd-trace-dotnet/pull/4884/commits/842f534625d1399fc42a8a5966aa5af9bbaef9df cc @tonyredondo